x/upgrade/README.md

```diff
@@ -1,23 +1,77 @@
 # `x/upgrade`
 
-## Abstract
+This upgrade module is a fork of the cosmos-sdk's [x/upgrade](https://github.com/cosmos/cosmos-sdk/tree/main/x/upgrade) module. The primary purpose of this module is to allow for rolling network upgrades as proposed in [ADR-018](../../docs/architecture/adr-018-network-upgrades.md) and [CIP-10](https://github.com/celestiaorg/CIPs/blob/main/cips/cip-10.md).
 
-This upgrade module is a fork of the cosmos-sdk's
-[x/upgrade](https://github.com/cosmos/cosmos-sdk/tree/main/x/upgrade) module
-that removes the entrypoints to the standard upgrade module by not registering a
-message server.
+Note: this module won't be used for upgrading from app version v1 to v2 but will be used for upgrading from v2 to v3 and onwards.
 
-The goal of this approach is to force social consensus to reach an upgrade
-instead of relying on token voting. It works by simply removing the ability of
-the gov module to schedule an upgrade. This way, the only way to upgrade the
-chain is to agree on the upgrade logic and the upgrade height offchain.
+## Concepts
 
-This fork registers the standard upgrade module types to preserve the ability to
-marshal them. Additionally the keeper of the standard upgrade module is still
-added to the application.
+- Total voting power: The sum of voting power for all validators.
+- Voting power threshold: The amount of voting power that needs to signal for a particular version for an upgrade to take place. This is a percentage of the total voting power (usually 5/6).
 
-## Resources
+## State
 
-1. <https://github.com/celestiaorg/celestia-app/pull/1491>
+This module persists a map in state from validator address to version that they are signalling for.
+
+## State Transitions
+
+The map from validator address to version is updated when a validator signals for a version (`SignalVersion`) and after an upgrade takes place (`ResetTally`).
+
+## Messages
+
+See [types/msgs.go](./types/msgs.go) for the message types.
+
+## Begin Block
+
+N/A
+
+## End Block
+
+N/A
+
+## Hooks
+
+N/A
+
+## Events
+
+N/A
+
+## Client
+
+### CLI
+
+```shell
+celestia-appd query upgrade tally
+celestia-appd tx upgrade signal
+celestia-appd tx upgrade try-upgrade
+```
+
+### gRPC
+
+```api
+celestia.upgrade.v1.Query/VersionTally
+```
+
+```shell
+grpcurl -plaintext localhost:9090 celestia.upgrade.v1.Query/VersionTally
+```
+
+## Params - list all module parameters, their types (in JSON) and examples
+
+N/A
+
+## Future Improvements
+
+N/A
+
+## Tests
+
+N/A
```

Comment:
Is it necessary to keep the sections that are N/A?

Reply:
Not strictly necessary in my opinion. Can definitely remove

Reply:
https://github.com/celestiaorg/celestia-app/pull/3224/commits/cbea0ebb070b29c9c04914fc039ac961886fd4e9